### ISSUE TITLE
Unify check_yum.py in original upstream project for easier maintenance (Advanced Nagios Plugins Collection)

### ISSUE THREAD

**Issue by HariSekhon** [calestyo/check_yum#27]
Hi @calestyo 

Instead of maintaining multiple divergent copies of this check_yum.py code, could you please just send any updates you want to [the original Advanced Nagios Plugins Collection repo](https://github.com/harisekhon/nagios-plugins) via a pull request?

This way there is less maintenance effort overall and everyone benefits from all of of our joint improvements (GitHub will also give you credit as a contributor to that larger higher profile project).

It doesn't really make sense for each of us to making updates for this same code independently, it's twice the work.

The upstream project also has layers of code checks, continuous integration, dockerized tests etc so the quality will be kept higher there and will be supported long term.

**Comment by calestyo** [comment]
Hey.

Well I took over check_yum from you some years ago, when you said you wouldn't have time for it any longer… you even gave your consent to this.

That being said, I haven't really had enough time for it myself in recent years, so I'd be more than happy to give maintainership back, if you're willing to take it over officially.
In that case I'd also give you ownership on https://exchange.nagios.org/directory/Plugins/Operating-Systems/Linux/check_yum/details

Though I think it would be more beneficial if this is kept as a separate repo… your nagios-plugins repo seems to be collection of n different checks, all basically independent of each other.
Mixing that many programs in one repo not only makes the repo much less easy to administrate, but the whole thing much difficult to package/etc.


Cheers,
Chris.

**Comment by HariSekhon** [comment]
Yes I remember that was a hectic time you caught me at, thanks very much for giving this plugin some maintenance.

Around 2012/2013 I pulled them all my plugins back in to one repo with greater focus - the project is in a good state in terms of both fix support and supporting infrastructure (CI, Dockerized multi-version test harnesses, pre-built Docker containers available on DockerHub with all dependencies included etc). It's been a lot of ongoing work but I'm committed to not just maintaining it but driving it forwards as I have a huge backlog of awesome improvements and new plugins inbound, especially as it's been well received and has good momentum.

You're welcome to submit GitHub pull requests to the upstream project.

Thanks

Hari

**Comment by calestyo** [comment]
Well I think the best solution is, if you take over maintenance and we move the upstream location to your repo (though I still think that having such multi-project-repos is a bad idea per se).

I've already changed the URL at https://exchange.nagios.org/directory/Plugins/Operating-Systems/Linux/check_yum/details ... could you please go there and "claim" the plugin entry? And then add URLs/email addresses/etc. as you wish.

**Comment by calestyo** [comment]
As for this repo and the last version of check_yum in it... your's (https://github.com/HariSekhon/nagios-plugins/blob/master/check_yum.py) seems to be the very old (and IIRC dysfunctional) version from back then when I took over.

If you want to keep everything in that one big repo, then you should probably merge the most recent version from here (possibly with some open pull requests if you like them).
Even if you keep your all-in-one-repo it should be easily possible to pull the commits from this one with all the history into it.

Alternatively you could simply clone this repo and let that be the new canonical one.

**Comment by calestyo** [comment]
Or, I could transfer you this repo (if you like to keep the issue tracker)... if not I'd add comments to all open issues / pull requests, notifying people that there is a new maintainer an new upstream repo.

What would you prefer?

**Comment by HariSekhon** [comment]
The version at the upstream repo has received ongoing updates, and has tests.

It might be worth just putting a README.md file in the repo to make a note to also see the upstream repo.
